generator/internal/discovery_proto_export_file_test.cc

```diff
@@ -0,0 +1,80 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "generator/internal/discovery_proto_export_file.h"
+#include "google/cloud/testing_util/status_matchers.h"
+#include <gmock/gmock.h>
+
+namespace google {
+namespace cloud {
+namespace generator_internal {
+namespace {
+
+using ::testing::Eq;
+
+TEST(DiscoveryProtoExportFileTest, FormatFile) {
+  auto constexpr kExpectedHeaderFile = R"""(// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Generated by the Codegen C++ plugin.
+// If you make any local changes, they will be lost.
+
+// The contents of internal/common_xxx.proto files are subject to change, and it
+// is not recommended to include their respective headers directly. Including
+// the <service>.pb.h file transitively provides all the necessary proto
+// definitions. However, it is not IWYU friendly. Including this header file
+// uses IWYU pragmas to export the internal/common_xxx.pb.h files as an
+// IWYU-friendly alternative.
+
+#ifndef GOOGLE_CLOUD_CPP_RELATIVE_FILE_PATH
+#define GOOGLE_CLOUD_CPP_RELATIVE_FILE_PATH
+
+// IWYU pragma: begin_exports
+#include <google/cloud/compute/v1/internal/common_001.pb.h>
+#include <google/cloud/compute/v1/internal/common_002.pb.h>
+// IWYU pragma: end_exports
+
+#endif  // GOOGLE_CLOUD_CPP_RELATIVE_FILE_PATH
+)""";
+
+  DiscoveryProtoExportFile f(
+      "my_output_file_path", "relative_file_path",
+      {"google/cloud/compute/v1/internal/common_001.proto",
+       "google/cloud/compute/v1/internal/common_002.proto"});
+  std::stringstream os;
+  auto result = f.FormatFile(os);
+  ASSERT_STATUS_OK(result);
+  EXPECT_THAT(os.str(), Eq(kExpectedHeaderFile));
+}
+
+}  // namespace
+}  // namespace generator_internal
+}  // namespace cloud
+}  // namespace google
+
+int main(int argc, char** argv) {
+  ::testing::InitGoogleTest(&argc, argv);
+  return RUN_ALL_TESTS();
+}
```

Comment:
I don't know why the "google_cloud_cpp_generator_unit_tests" don't use `gtest_main`.